packages/pwa/app/utils/url.js

```diff
@@ -5,9 +5,26 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {DEFAULT_LOCALE} from '../constants'
-import {getUrlConfig} from './utils'
+import {DEFAULT_SITE_ID, HOME_HREF} from '../constants'
+import {getConfig} from './utils'
 import {urlPartPositions} from '../constants'
+import {getAppOrigin} from 'pwa-kit-react-sdk/utils/url'
+
+/**
+ * A function that takes a path and qualifies it with the current host and protocol.
+ * This function works on the client and on the server.
+ *
+ * @example
+ * urlToPath(/women/dresses?color=black)
+ *
+ * // returns //http(s)://www.site.com/women/dresses?color=black
+ * @param path
+ * @returns {string|*}
+ */
+export const pathToUrl = (path) => {
+    const url = typeof window === 'undefined' ? `${getAppOrigin()}${path}` : window.location.href
```

Comment:
You code doesn't strictly follow what you are describing. On the client you are actually returning the href full stop and not qualifying the path with the origin like you are saying you are going to do. You should do this below 👇 

```suggestion
    const url = typeof window === 'undefined' ? `${getAppOrigin()}${path}` : `${window.location.origin}${path}`
```